cprover_bindings/src/irep/goto_binary_serde.rs

```diff
@@ -0,0 +1,1462 @@
+        GotoBinarySharingStats::from_deserializer(self)
+    }
+
+    /// Returns Err if the found value is not the expected value.
+    fn expect<T: Eq + std::fmt::Display>(found: T, expected: T) -> io::Result<T> {
+        if found != expected {
+            return Err(Error::new(
+                ErrorKind::Other,
+                format!("expected {expected} in byte stream, found {found} instead)"),
+            ));
+        }
+        Ok(found)
+    }
+
+    /// Adds an InternedString unique number to the "read" cache, returns true
+    /// iff was never read before.
+    fn is_first_read_string(&mut self, u: usize) -> bool {
+        if u >= self.string_count.len() {
+            self.string_count.resize(u + 1, 0);
+        }
+        let count = self.string_count[u];
+        self.string_count[u] = count.checked_add(1).unwrap();
+        count == 0
+    }
+
+    /// Maps a string number used in the byte stream to the number generated by
+    /// our own numbering for that string.
+    fn add_string_mapping(&mut self, num_binary: usize, num: usize) {
+        if num_binary >= self.string_map.len() {
+            self.string_map.resize(num_binary + 1, None);
+        }
+        let old = self.string_map[num_binary];
+        if old.is_some() {
+            panic!("string number already mapped");
+        }
+        self.string_map[num_binary] = Some(num);
+    }
+
+    /// Adds an Irep unique number to the "read" cache, returns true iff it was
+    /// never read before.
+    fn is_first_read_irep(&mut self, u: usize) -> bool {
+        if u >= self.irep_count.len() {
+            self.irep_count.resize(u + 1, 0);
+        }
+        let count = self.irep_count[u];
+        self.irep_count[u] = count.checked_add(1).unwrap();
+        count == 0
+    }
+
+    /// Maps an Irep number used in the byte stream to the number generated by
+    /// our own numbering for that Irep.
+    fn add_irep_mapping(&mut self, num_binary: usize, num: usize) {
+        if num_binary >= self.irep_map.len() {
+            self.irep_map.resize(num_binary + 1, None);
+        }
+        let old = self.irep_map[num_binary];
+        if old.is_some() {
+            panic!("irep number already mapped");
+        }
+        self.irep_map[num_binary] = Some(num);
+    }
+
+    /// Reads a u8 from the byte stream.
+    fn read_u8(&mut self) -> io::Result<u8> {
+        match self.bytes.next() {
+            Some(Ok(u)) => Ok(u),
+            Some(Err(error)) => Err(error),
+            None => Err(Error::new(ErrorKind::Other, "unexpected end of input")),
+        }
+    }
+
+    /// Reads a usize from the byte stream assuming it is encoded using 7-bi
+    /// variable length encoding ([GotoBinarySerializer::write_usize_varenc]).
+    fn read_usize_varenc(&mut self) -> io::Result<usize> {
+        let mut result: usize = 0;
+        let mut shift: usize = 0;
+        let max_shift: usize = std::mem::size_of::<usize>() * std::mem::size_of::<u8>() * 8;
```

Comment:
Should this just be:
```suggestion
        let max_shift: usize = std::mem::size_of::<usize>() * 8;
```
?

Reply:
`std::mem::size_of::<u8>()` is equal to one, removing it